### PR TITLE
Update Ray Argument

### PR DESCRIPTION
**Pull request by donglihe-hub**
## What does this PR do?

1. Add shorthand for --config
2. Since 2.24.0, using local_dir in ray.tune will raise an error. a minimum version requirement is set for ray[tune]. The first deprecation comment occurs in 2.5.0.
3. Upgrade ray.tune API from 1.0 style to 2.0 because the documentation of the 1.0-style API is no longer updated.

image

image

**Results:**
Before:
image

image

After:
image
image

## Test CLI & API (`bash tests/autotest.sh`)
Test APIs used by main.py.
- [ ] Test Pass
  - (Copy and paste the last outputted line here.)
- [x] Not Applicable (i.e., the PR does not include API changes.)

## Check API Document
If any new APIs are added, please check if the description of the APIs is added to API document. 
- [ ] API document is updated ([linear](https://www.csie.ntu.edu.tw/~cjlin/libmultilabel/api/linear.html), [nn](https://www.csie.ntu.edu.tw/~cjlin/libmultilabel/api/nn.html))
- [x] Not Applicable (i.e., the PR does not include API changes.)

## Test quickstart & API (`bash tests/docs/test_changed_document.sh`)
If any APIs in quickstarts or tutorials are modified, please run this test to check if the current examples can run correctly after the modified APIs are released.

### CODE REVIEW

**Review by Eleven1Liu — CHANGES_REQUESTED** [review]
Please help add `ray[tune] > ???` to [requirements_parameter_search.txt
](https://github.com/ASUS-AICS/LibMultiLabel/blob/801733845386bb91c90f8b81abac394bde6c7d85/requirements_parameter_search.txt#L3)

File: search_params.py [hunk 1]
```diff
@@ -141,7 +141,10 @@ def init_search_algorithm(search_alg, metric=None, mode=None):
         from ray.tune.search.bayesopt import BayesOptSearch
 
         return BayesOptSearch(metric=metric, mode=mode)
-    logging.info(f"{search_alg} search is found, run BasicVariantGenerator().")
+    elif search_alg == "basic_variant":
+        pass
+    else:
+        logging.info(f"No Search algorithm is found. Will run basic_variant")
```
Comment (Eleven1Liu) [inline search_params.py:147]: Why add these codes?

**Review by Eleven1Liu — CHANGES_REQUESTED** [review]

File: search_params.py [hunk 2]
```diff
@@ -171,13 +174,11 @@ def prepare_retrain_config(best_config, best_log_dir, retrain):
         best_config.merge_train_val = False
 
 
-def load_static_data(config, merge_train_val=False):
```
Comment (Eleven1Liu) [inline search_params.py:174]: Are these code changes necessary in this PR?

**Review by Eleven1Liu — CHANGES_REQUESTED** [review]

File: search_params.py [hunk 3]
```diff
@@ -231,7 +232,7 @@ def retrain_best_model(exp_name, best_config, best_log_dir, retrain):
     with open(os.path.join(checkpoint_dir, "params.yml"), "w") as fp:
         yaml.dump(dict(best_config), fp)
 
-    data = load_static_data(best_config, merge_train_val=best_config.merge_train_val)
+    data = load_static_data(best_config)
```
Comment (Eleven1Liu) [inline search_params.py:235]: Same as the previous comment. Why code changes here?

File: search_params.py (hunk 1 above)
Comment (donglihe-hub) [reply search_params.py:147]: The former logic could print weird stuff if user inputs something like "abcdefg"

Comment (donglihe-hub) [reply search_params.py:147]: I forgot to change this part... My original thought was to inform users of the search algorithm they use...

File: search_params.py (hunk 2 above)
Comment (donglihe-hub) [reply search_params.py:174]: Not really...But the argument has already be included in config

File: search_params.py (hunk 3 above)
Comment (donglihe-hub) [reply search_params.py:235]: Not really...But the argument has already be included in config

**Review by Eleven1Liu — APPROVED** [review]

**Pull request merged**